### ISSUE TITLE
[dev.icinga.com #8100] POSTGRES IDO: invalid syntax for integer: "true" while trying to update table icinga_hoststatus

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#2442]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/8100

**Created by harvey on 2014-12-16 15:57:07 +00:00**

Assignee: _gbeutner_
Status: _Resolved_  (closed on _2014-12-17 08:00:03 +00:00_)
Target Version: _2.3.0_
Last Update: _2015-02-13 07:02:34 +00:00 (in Redmine)_

    Icinga Version: icinga2-2.2.2-1.snapshot201412161449.el6.x86_64
    Backport?: No

---

Hi, after icinga2 updates from 2014-12-15 and 2014-12-16 icinga2 tries to update table "icinga\_hoststatus" column "should\_be\_scheduled" with a value E"true", where it schould be integer.

    [2014-12-16 16:48:49 +0100] warning/IdoPgsqlConnection: Exception during database operation: Verify that your database is operational!
    [2014-12-16 16:48:58 +0100] information/IdoPgsqlConnection: pgSQL IDO instance id: 1 (schema version: '1.12.0')
    [2014-12-16 16:48:59 +0100] critical/IdoPgsqlConnection: Error "ERROR:  invalid input syntax for integer: "true"
    LINE 1: ...led_downtime_depth = E'0',  should_be_scheduled = E'true',  ...
                                                                                                                  ^
    " when executing query "UPDATE icinga_hoststatus SET acknowledgement_type = E'0',  active_checks_enabled = E'1',  check_command = E'hostalive',  check_source = E'icinga',  check_type = E'0',  current_check_attempt = E'1',  current_notification_number = E'0',  current_state = E'0',  endpoint_object_id = 1,  event_handler = E'',  event_handler_enabled = E'0',  execution_time = E'4.0520310401916504',  flap_detection_enabled = E'1',  has_been_checked = E'1',  host_object_id = 176,  instance_id = 1,  is_flapping = E'0',  is_reachable = E'1',  last_check = TO_TIMESTAMP(1418744893),  last_hard_state = E'0',  last_hard_state_change = TO_TIMESTAMP(1418041886),  last_state_change = TO_TIMESTAMP(1418041856),  last_time_down = TO_TIMESTAMP(1414773079),  last_time_unreachable = TO_TIMESTAMP(1414773136),  last_time_up = TO_TIMESTAMP(1418744893),  latency = E'0',  long_output = E'',  max_check_attempts = E'5',  modified_host_attributes = E'0',  next_check = TO_TIMESTAMP(1418744949),  normal_check_interval = E'1',  notifications_enabled = E'1',  output = E'PING OK - Packet loss = 0%, RTA = 15.98 ms',  passive_checks_enabled = E'1',  percent_state_change = E'0',  perfdata = E'rta=15.983000ms;3000.000000;5000.000000;0.000000 pl=0%;80;100;0',  problem_has_been_acknowledged = E'0',  process_performance_data = E'0',  retry_check_interval = E'0.5',  scheduled_downtime_depth = E'0',  should_be_scheduled = E'true',  state_type = E'1',  status_update_time = TO_TIMESTAMP(1418744939) WHERE host_object_id = 176"
    Context:
            (0) Reconnecting to PostgreSQL IDO database 'ido-pgsql'

Nothing changed around that, only syntax errors were corrected:
old: assign where host.name  "store-4" & service.name  "ssh"
new: assign where host.name  "store-4" && service.name  "ssh"

icinga2 was working up to the update from 2014-12-15.

Database table icinga\_hoststatus has integer configured.

    ...
    has_been_checked    integer NULL [0]     
    should_be_scheduled integer NULL [0]     
    current_check_attempt   integer NULL [0]     
    max_check_attempts  integer NULL [0]
    ...

**Changesets**

_2014-12-16 19:40:44 +00:00_ by _mfriedrich_ 6299f07914846749a16b294a4106c9540fda2711

    DB IDO: Fix 'should_be_scheduled' status table column
    
    refs #8100

_2014-12-16 20:23:50 +00:00_ by _gbeutner_ 884c7053d7b9fefffc2bfa738faf43b754cda023

    Fix incorrect SQL values for boolean fields
    
    fixes #8100

_2014-12-16 20:28:13 +00:00_ by _gbeutner_ 5e1182a5c79a6b8d43c7098ec864dbd4b7ad22c8

    Fix incorrect SQL values for boolean fields
    
    fixes #8100

_2014-12-16 20:43:03 +00:00_ by _gbeutner_ 5dd0bbfe316c841a1ecee2f5ffd3e266c305c23d

    Fix another problem with boolean values in SQL queries
    
    refs #8100

_2014-12-17 07:54:28 +00:00_ by _gbeutner_ 9eeb64a7805074ac0cda67f68b690a2b9b0cc282

    Update livestatus and statusdata to properly support booleans
    
    fixes #8100

---

**Relations**:

* relates [#8043](https://dev.icinga.com/issues/8043)
